File: boj/swift/1021.swift
//
//  1021번: 회전하는 큐
//  https://www.acmicpc.net/problem/1021
//
//  Created by WhiteHyun on 2021/11/20.
//

import Foundation


let size: Int = Int(readLine()!.components(separatedBy: " ")[0])!
let inputArray: [Int] = readLine()!.components(separatedBy: " ").map { Int(String($0))! }
var dequeArray: [Int] = [Int](1...size) // 여기서 size는 10

var position: Int = 0 // 덱의 위치 (인덱스)
var minimum: Int = 0

for number in inputArray {
    
    let index: Int = dequeArray.firstIndex(of: number)!
    minimum += min(abs(position - index), dequeArray.count - abs(position - index))
    position = dequeArray.count == index ? 0 : index
    dequeArray.remove(at: index)
}

print(minimum)

File: boj/python/1021.py
# 1021번: 회전하는 큐
# https://www.acmicpc.net/problem/1021
# Version: Python 3.8.12


from sys import stdin

read = stdin.readline

if __name__ == "__main__":
    size, _ = map(int, read().split())
    input_list = list(map(int, read().split()))
    deque = [i for i in range(1, size + 1)]
    minimum = 0  # 최소 이동값
    position = 0  # 현재 위치

    for number in input_list:
        index = deque.index(number)
        minimum += min(abs(index - position), len(deque) - abs(index - position))
        deque.pop(index)
        position = index if index < len(deque) else 0

    print(minimum)
